src/supervisor.sbvr

```diff
@@ -0,0 +1,461 @@
+		Necessity: each uuid is of exactly one device.
+		Necessity: each device has a uuid that has a Length (Type) that is equal to 32.
+
+	Fact type: device has hostname
+		Necessity: each device has exactly one hostname.
+
+	Fact type: device has proxy
+		Necessity: each device has at most one proxy.
+
+	Fact type: device is connected to vpn
+
+
+
+-- DEVICE METRICS
+
+Term: cpu id
+	Concept Type: Short Text (Type)
+
+Term: cpu temp
+	Concept Type: Integer (Type)
+
+Term: cpu usage
+	Concept Type: Integer (Type)
+
+Term: memory total
+	Concept Type: Integer (Type)
+
+Term: memory usage
+	Concept Type: Integer (Type)
+
+Term: storage block device
+	Concept Type: Short Text (Type)
+
+Term: storage total
+	Concept Type: Integer (Type)
+
+Term: storage usage
+	Concept Type: Integer (Type)
+
+Term: metric
+-- Separate metrics from other device info for greater modularity
+-- Not sure about syntax/accuracy/necessity for this section, needs sanity check
+
+	Fact type: device provides metric
+	-- Is this necessary?
+		Synonymous Form: metric is provided by device
+		Necessity: each metric is of at most one device
+		Necessity: each device provides one or more metrics
+
+	Fact type: metric has cpu id
+		Synonymous Form: device provides cpu id
+		Necessity: each metric has at most one cpu id
+
+	Fact type: metric has cpu temp
+		Synonymous Form: device provides cpu temp
+		Necessity: each metric has at most one cpu temp
+
+	Fact type: metric has cpu usage
+		Synonymous Form: device provides cpu usage
+		Necessity: each metric has at most one cpu usage
+
+	Fact type: metric has memory usage
+		Synonymous Form: device provides memory usage
+		Necessity: each metric has at most one memory usage
+
+	Fact type: metric has memory total
+		Synonymous Form: device provides memory total
+		Necessity: each metric has at most one memory total
+
+	Fact type: metric has storage block device
+		Synonymous Form: device provides storage block device
+		Necessity: each metric has at most one storage block device
+
+	Fact type: metric has storage usage
+	 	Synonymous Form: device provides storage usage
+		Necessity: each metric has at most one storage usage
+
+	Fact type: metric has storage total
+	 	Synonymous Form: device provides storage total
+		Necessity: each metric has at most one storage total
+
+	Fact type: device is undervolted
+		Synonymous form: metric is undervolted
+			-- "metric is undervolted" doesn't make sense but not sure how to tie a boolean value to a sub-category (metric is a subcategory of device)
+
+
+
+-- IMAGE
+
+Term: image location
+	Concept Type: Short Text (Type)
+
+Term: image
+
+	Fact type: image is stored at image location
+		Synonymous Form: image location hosts image
+		Necessity: each image is stored at exactly one image location.
+		Necessity: each image location hosts exactly one image.
+
+	Fact type: image has content hash
+		Necessity: each image has at most one content hash.
+
+	Fact type: image is downloaded by device
```

Comment:
This is confusing for me, I think for the device we need to distinguish between target images (i.e. the description of an image hosted on the cloud), and current images (what's installed on the device right now). I'm not sure how to solve that with the model, but I think is probably something we need to investigate. 

The same applies for all entities, app, release, etc.

Reply:
The relationship should probably be between app and device, app has a target release and release has images (or something like that)